src/core/tests/queryObserver.test.tsx

```diff
@@ -143,7 +144,7 @@ describe('queryObserver', () => {
     await sleep(1)
     await observer.refetch()
     unsubscribe()
-    expect(count).toBe(2)
+    expect(count).toBe(3)
```

Comment:
If the query function returned a constant value, then only the first call would have any effect because of the bail out change. So now it returns a different value on each call. This results in one extra call to `select`.